src/Sylius/Component/Core/Statistics/Provider/SalesStatisticsProvider.php

```diff
@@ -63,4 +111,52 @@ private function getPeriodFormat(string $intervalType): string
 
         return $this->formatsMap[$intervalType];
     }
+
+    /**
+     * @param array<array{period: \DateTimeInterface, ...}> $firstArray
+     * @param array<array{period: \DateTimeInterface, ...}> $secondArray
+     *
+     * @return array<array{period: \DateTimeInterface, ...}>
+     */
+    private function mergeArraysByPeriod(array $firstArray, array $secondArray): array
+    {
+        $indexByPeriod = function(array $items) {
+            $result = [];
+            foreach ($items as $item) {
+                if (!isset($item['period']) || !$item['period'] instanceof \DateTimeInterface) {
+                    continue;
+                }
+                $key = $item['period']->format('c');
+                $result[$key] = $item;
+            }
+
+            return $result;
+        };
+
+        $indexedFirstArray = $indexByPeriod($firstArray);
+        $indexedSecondArray = $indexByPeriod($secondArray);
+
+        $allKeys = array_unique(array_merge(array_keys($indexedFirstArray), array_keys($indexedSecondArray)));
```

Comment:
This array could be populated inside the `$indexByPeriod`;
As it is we have `array_unique(array_merge(...` inside a loop, and both are pretty costly